flow360/version_check.py

```diff
@@ -0,0 +1,128 @@
+"""
+This module provides utility functions for handling client versions.
+
+Classes:
+- VersionSupported: Enumeration representing the different states of version support.
+
+Functions:
+- get_supported_server_versions(app_name: str) -> List[str]: Fetches a list of supported versions from the server.
+- check_client_version(app_name: str) -> Tuple[VersionSupported, Union[str, Version]]: 
+    Checks the current client version against the available versions.
+- client_version_get_info(app_name: str) -> None: Logs information about the client version.
+
+Constants:
+- YES: VersionSupported enum value indicating that the version is supported.
+- NO: VersionSupported enum value indicating that the version is no longer supported.
+- CAN_UPGRADE: VersionSupported enum value indicating that an upgrade is available.
+
+"""
+import re
+from enum import Enum
+from typing import List, Tuple, Union
+
+import requests
+from packaging.version import Version
+from requests import HTTPError
+
+from .cloud import http_util
+from .log import log
+from .solver_version import Flow360Version as sv
+from .version import __version__
+
+
+class VersionSupported(Enum):
+    """
+    Enumeration representing the different states of version support.
+    """
+
+    YES = 1
+    NO = 2
+    CAN_UPGRADE = 3
+
+
+def get_supported_server_versions(app_name: str) -> List[str]:
+    """
+    Fetch a list of supported versions from the server
+
+    Args:
+        app_name (str): The name of the application
+
+    Returns:
+        List[str]: A list of supported version strings
+    """
+    try:
+        session = requests.Session()
+        http = http_util.Http(session)
+        response = http.get(f"versions?app_name={app_name}")
+    except HTTPError as error:
+        raise HTTPError(f"failed to retrieve the versions for {app_name}") from error
+
+    versions = [re.sub(r".+-", "", item["version"]) for item in response]
+
+    if len(versions) == 0:
+        raise RuntimeError(f"the supported versions should not be empty for {app_name}")
```

Comment:
Lets rewrite this massage:
f"Something went wrong when checking python client version. The supported versions should not be empty for {app_name}. If you see this message again, contact support."